src/components/toggles/toggleGroup/toggleGroup.tsx

```diff
@@ -0,0 +1,44 @@
+import { ToggleGroup as RadixToggleGroup } from '@radix-ui/react-toggle-group';
+import classNames from 'classnames';
+import type { ComponentProps } from 'react';
+
+export type ToggleGroupValue<TMulti extends boolean> = TMulti extends true ? string[] | undefined : string | undefined;
+
+export interface IToggleGroupBaseProps<TMulti extends boolean>
+    extends Omit<ComponentProps<'div'>, 'value' | 'onChange' | 'defaultValue' | 'ref' | 'dir'> {
+    /**
+     * Allows multiple toggles to be selected at the same time when set to true.
+     */
+    isMultiSelect: TMulti;
```

Comment:
That would be nice, I tried setting it as optional but TypeScript complains on the `single` return. Somehow it does not infer the correct type (`string | undefined`) for the value prop. Also the conditional return on the `ToggleGroup` component and this complex type:
```
export type ToggleGroupValue<TMulti extends boolean> = TMulti extends true ? string[] | undefined : string | undefined;
```
Is needed because radix-ui does not accept a variable on the `type` props, it's either a fixed `multiple` or `single` value :/

Reply:
It's a type narrowing thing I think. A quick workaround might be explicitly cast the falsy path or return null for the undefined one. Thoughts?

Reply:
I'd prefer keeping the field as required rather than introduce a dirty fix to be honest 😅